endpoints/openrtb2/video_auction.go

```diff
@@ -79,7 +81,27 @@ func (deps *endpointDeps) VideoAuctionEndpoint(w http.ResponseWriter, r *http.Re
 		CookieFlag:    pbsmetrics.CookieFlagUnknown,
 		RequestStatus: pbsmetrics.RequestStatusOK,
 	}
+
+	debugQuery := r.URL.Query().Get("debug")
+	enableDebug := debugQuery == "true"
+	var debugString string
+	var errorCacheID string
+
 	defer func() {
+		if len(errorCacheID) > 0 && vo.VideoResponse == nil {
+			cache := deps.ex.GetCache()
+			debugString = fmt.Sprintf("<!--\n%s\n\nNo response created\n-->", debugString)
+			data, _ := json.Marshal(debugString)
+			toCache := []prebid_cache_client.Cacheable{
+				{
+					Type:       prebid_cache_client.TypeXML,
+					Data:       data,
+					TTLSeconds: int64(deps.cfg.CacheURL.DefaultTTLs.Video),
+					Key:        fmt.Sprintf("log_%s", errorCacheID),
+				},
+			}
+			cache.PutJson(context.Background(), toCache)
```

Comment:
Please use a timed context here and consider exposing the duration as a config option, or perhaps since this is temporary code until a better logging mechanism can be built just use the same 50ms hardcoded value as the auction reserve time.

Reply:
Updated to use a deadline. I used the `defaultRequestTimeout` defined in the Video auction. As a temporary solution, I am not sure if I should add a config value but I am open to it if it better defined there.

Reply:
The `defaultRequestTimeout` is 5 seconds. I think that's too long to wait for a cache call. I think you can just hardcode it for 50ms to match the hardcoded elsewhere. Honestly, since those are non-configurable this one is probably fine with a hardcoded value as well.. at least for now.

Reply:
It seems that at least running the server locally, the cache call here is exceeding 50ms. 100ms seems to work okay. Should it be definitely less than 50ms or is somewhere around 100ms for the timeout appropriate?

Reply:
50ms the minimal cache reserve time for AMP and is usually not an issue. I don't want to split hairs over 50ms vs 100ms, especially for a debugging only use case. 100ms would be fine.

Reply:
Understood! I will keep at 100ms for the time being.